src/Common.cpp

```diff
@@ -177,13 +177,27 @@ std::vector<std::string> NAS2D::split(std::string str, char delim /*= ','*/, boo
 std::pair<std::string, std::string> NAS2D::splitOnFirst(const std::string& str, char delim)
 {
 	const auto delim_loc = str.find_first_of(delim);
-	return std::make_pair(str.substr(0, delim_loc), str.substr(delim_loc + 1));
+	if (delim_loc == std::string::npos)
+	{
+		return std::make_pair(std::string{}, str);
+	}
+	else
+	{
+		return std::make_pair(str.substr(0, delim_loc), str.substr(delim_loc + 1));
+	}
 }
 
 std::pair<std::string, std::string> NAS2D::splitOnLast(const std::string& str, char delim)
 {
 	const auto delim_loc = str.find_last_of(delim);
-	return std::make_pair(str.substr(0, delim_loc), str.substr(delim_loc + 1));
+	if (delim_loc == std::string::npos)
+	{
+		return std::make_pair(str, std::string{});
```

Comment:
This one also feels reversed to my expectations.

Reply:
Now I'm confused. You said the previous one felt backwards, yet this is the reverse of the previous and you're saying it's backwards too.

Reply:
```cpp
splitOnLast("abc,", ',');  // { "abc", "" }
splitOnLast("ab,c", ',');  // { "ab", "c" }
splitOnLast("a,bc", ',');  // { "a", "bc" }
splitOnLast(",abc", ',');  // { "", "abc" }
splitOnLast("abc", ',');  // { "", "abc" }  <- Suggestion (rather than { "abc", "" })
```

If the second component is all the stuff after the delimiter (scanning backwards), and there is no delimiter, I feel the second component should then be the whole string.